src/lib/components/composites/settings/user-settings/ChangeNameSettings.svelte

```diff
@@ -0,0 +1,111 @@
+    import { Schema } from "$lib/schemas";
+    import { toast } from "svelte-sonner";
+    import Input from "../../input/Input.svelte";
+    import ErrorAlert from "../../utils/ErrorAlert.svelte";
+    import SettingsSection from "../SettingsSection.svelte";
+    import { generateFieldMask } from "protobuf-fieldmask";
+
+    interface Props {
+        user: User;
+    }
+
+    const { user }: Props = $props();
+
+    let firstNameInput: Input;
+    let lastNameInput: Input;
+
+    let updateUserError: ApiError | undefined = $state(undefined);
+
+    async function handleSubmit(event: Event) {
+        event.preventDefault();
+
+        const isFirstNameValid = firstNameInput.validate();
+        const isLastNameValid = lastNameInput.validate();
+        if (!(isFirstNameValid && isLastNameValid)) return;
+
+        const userData: Partial<User> = {
+            id: user.id,
+            firstName: firstNameInput.getValue(),
+            lastName: lastNameInput.getValue(),
+        };
+
+        const updatedUser = User.create(userData);
+        const maskPaths = generateFieldMask(userData).filter((path) => path !== "id");
+
+        backendService
+            .updateUser({
+                user: updatedUser,
+                mask: {
+                    paths: maskPaths,
+                },
+            })
+            .response.then(() => {
+                invalidate("data:getCurrentUser");
+                toast("Successfully updated your name.");
+
+                // Make sure the input fields are not focused after submitting
+                (document.activeElement as HTMLElement)?.blur();
+            })
+            .catch((error) => {
+                updateUserError = { errorTitle: "Something went wrong while updating user." };
+                console.error(error);
+            });
+    }
+</script>
+
+<!--
+@component
+Component for changing the user's name.
+
+Usage:
+```svelte
+    <ChangeNameSettings {user} />
+```
+-->
+<SettingsSection sectionTitle="Change Name">
+    <p class="text-base">
+        Your name is displayed in the member list of projects and when hovering over your character.
+    </p>
+    <form
+        class="flex w-full max-w-100 flex-col items-center gap-2.5 md:h-fit md:max-w-200 md:flex-row md:items-start"
+        onsubmit={handleSubmit}
+    >
+        <Input
+            bind:this={firstNameInput}
+            class="h-full w-full"
+            inputId="first-name-input"
+            label="First Name"
+            placeholder="First Name"
+            required
+            schema={Schema.firstName}
+            type="text"
+            value={user.firstName}
+        />
+        <Input
+            bind:this={lastNameInput}
+            class="h-full w-full"
+            inputId="last-name-input"
+            label="Last Name"
+            placeholder="Last Name"
+            required
+            schema={Schema.lastName}
+            type="text"
+            value={user.lastName}
+        />
```

Comment:
I nearly have overseen this. To make this component more reusable, we don't use an error message prefix by default. Please add this to get a sensible error message.
```suggestion
        <Input
            bind:this={firstNameInput}
            class="h-full w-full"
            inputId="first-name-input"
            label="First Name"
            placeholder="First Name"
            required
            schema={Schema.firstName}
            errorMessagePrefix="First Name must contain"
            type="text"
            value={user.firstName}
        />
        <Input
            bind:this={lastNameInput}
            class="h-full w-full"
            inputId="last-name-input"
            label="Last Name"
            placeholder="Last Name"
            required
            schema={Schema.lastName}
            errorMessagePrefix="Last Name must contain"
            type="text"
            value={user.lastName}
        />
```

Reply:
You're right, i didn't see that. I changed it!